arbitrator/validator/src/spawner_endpoints.rs

```diff
@@ -0,0 +1,79 @@
+// Copyright 2025-2026, Offchain Labs, Inc.
+// For license information, see https://github.com/OffchainLabs/nitro/blob/master/LICENSE.md
+
+//! Endpoints related to the `ValidationSpawner` Go interface and used by the nitro's validation
+//! client.
+//! Some of the data structures here are counterparts of Go structs defined in the `validator`
+//! package. Their serialization is configured to match the Go side (by using `PascalCase` for
+//! field names).
+
+use arbutil::{Bytes32, PreimageType};
+use axum::response::IntoResponse;
+use axum::Json;
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+
+pub async fn capacity() -> impl IntoResponse {
+    "1" // TODO: Figure out max number of workers (optionally, make it configurable)
+}
+
+pub async fn name() -> impl IntoResponse {
+    "Rust JIT validator"
+}
+
+pub async fn stylus_archs() -> impl IntoResponse {
+    if cfg!(target_os = "linux") {
+        if cfg!(target_arch = "aarch64") {
+            return "arm64";
+        } else if cfg!(target_arch = "x86_64") {
+            return "amd64";
+        }
+    }
+    "host"
+}
+
+pub async fn validate(Json(request): Json<ValidationRequest>) -> impl IntoResponse {
+    // TODO: Implement actual validation logic
+    serde_json::to_string(&request.start_state)
+        .map_err(|e| format!("Failed to serialize state: {e}",))
+}
+
+pub async fn wasm_module_roots() -> impl IntoResponse {
+    "[]" // TODO: Figure this out from local replay.wasm
+}
+
+type Hash = Bytes32;
+
+/// Counterpart for Go struct `validator.ValidationInput`.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(rename_all = "PascalCase")]
+pub struct ValidationRequest {
+    id: u64,
+    has_delayed_msg: bool,
+    #[serde(rename = "DelayedMsgNr")]
+    delayed_msg_number: u64,
+    preimages: HashMap<PreimageType, Hash>,
+    user_wasms: HashMap<String, HashMap<Hash, Vec<u8>>>,
+    batch_info: Vec<BatchInfo>,
+    delayed_msg: Vec<u8>,
+    start_state: GlobalState,
+    debug_chain: bool,
+}
+
+/// Counterpart for Go struct `validator.BatchInfo`.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(rename_all = "PascalCase")]
+pub struct BatchInfo {
+    number: u64,
+    data: Vec<u8>,
+}
+
+/// Counterpart for Go struct `validator.GoGlobalState`.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(rename_all = "PascalCase")]
+pub struct GlobalState {
+    block_hash: Hash,
+    send_root: Hash,
+    batch: u64,
+    pos_in_batch: u64,
+}
```

Comment:
Also shouldn't we avoid type alias's like `type Hash = Bytes32;`, you can already infer it is a hash by the field's name. Being able to read that the actual type is Bytes32 is very nice.

I think using type alias's for primitives like B256,u64 etc makes the code harder to read. Because then I need to check what the type alias is.

 For more complex type definitions like some type with generics it can be nice to use a type alias. If we want to keep the type alias it would be nice if it was instead below the imports above the structs/enums/impl/funcs at least that is where I expect to see them or

``use arbutil::{Bytes32 as Hash, PreimageType};``
^ do this

But I think this type alias makes the code harder to read in this case, so we should remove it, but up to you.

Reply:
B256/Bytes32 is almost always hashes as well, I can't think of an example at the time of my head where you would need it other then hashes

Reply:
Yeah I agree with only type aliasing complex types.